frontend/components/CellInput.js

```diff
@@ -828,6 +831,51 @@ export const CellInput = ({
         }
     }, [cm_forced_focus])
 
+    const on_explain = useCallback(async () => {
+        const code = newcm_ref.current?.state?.doc?.toString?.()
+        const key = pluto_actions.get_openai_key()
+
+        if (code && key) {
+            const response = await fetch("https://api.openai.com/v1/completions", {
+                method: "POST",
+                headers: {
+                    "Content-Type": "application/json",
+                    "Authorization": `Bearer ${key}`,
+                },
+                body: JSON.stringify({
+                    model: "text-davinci-003",
```

Comment:
If you use `gpt-3.5-turbo` then that's cheaper and more accurate. The API call is slightly different because it's via the `chat` endpoint, but it should work for now and also for the people who have GPT-4 access. Also, maybe allow people to set the endpoint because things are moving fast in OpenAI-land?